modmesh/onedim/euler1d.py

```diff
@@ -318,26 +318,31 @@ def build_field(self, t, coord=None):
         :return: None
         """
         if coord is None:
-            # set the x-coordinate for numerical solver.
+            # Set the x-coordinate for numerical solver.
             coord = self.svr.coord[self.svr.xindices]
-        self.coord = coord.copy()  # Make a copy; no write back to argument.
+        # Make a copy; no write back to argument.
+        self.coord_field = coord.copy()
 
         # Determine the zone location and the Boolean selection arrays.
         loc12, loc23, loc34, loc45 = self.calc_locations(t=t)
-        slct1 = self.coord < loc12
-        slct2 = np.logical_and(self.coord >= loc12, self.coord < loc23)
-        slct3 = np.logical_and(self.coord >= loc23, self.coord < loc34)
-        slct4 = np.logical_and(self.coord >= loc34, self.coord < loc45)
-        slct5 = self.coord >= loc45
+        slct1 = self.coord_field < loc12
+        slct2 = np.logical_and(self.coord_field >= loc12,
+                               self.coord_field < loc23)
+        slct3 = np.logical_and(self.coord_field >= loc23,
+                               self.coord_field < loc34)
+        slct4 = np.logical_and(self.coord_field >= loc34,
+                               self.coord_field < loc45)
+        slct5 = self.coord_field >= loc45
 
         # Create the field buffers.
-        self.density_field = np.zeros(self.coord.shape, dtype='float64')
-        self.velocity_field = np.zeros(self.coord.shape, dtype='float64')
-        self.pressure_field = np.zeros(self.coord.shape, dtype='float64')
-        self.temperature_field = np.zeros(self.coord.shape, dtype='float64')
-        self.internal_energy_field = np.zeros(self.coord.shape,
+        self.density_field = np.zeros(self.coord_field.shape, dtype='float64')
```

Comment:
@Gene0315 Please replace `np.zeros(...)` with `np.zeros_like(...)` and keep `dtype="..."` explicit.  See [`numpy.zeros_like()`](https://numpy.org/doc/2.2/reference/generated/numpy.zeros_like.html).